tests/test_utils/gcp_system_decorator.py

```diff
@@ -0,0 +1,194 @@
+* Run this test within automated environment variable workspace where
+  config directory is checked out next to the airflow one.
+
+""".format(
+    __file__
+)
+
+SKIP_LONG_TEST_WARNING = """
+The test is only run when the test is run in with GCP-system-tests enabled
+environment. And environment variable GCP_ENABLE_LONG_TESTS is set to True.
+You can enable it in one of two ways:
+
+* Set GCP_CONFIG_DIR environment variable to point to the GCP configuration
+  directory which keeps variables.env file with environment variables to set
+  and keys directory which keeps service account keys in .json format and
+  set GCP_ENABLE_LONG_TESTS to True
+* Run this test within automated environment variable workspace where
+  config directory is checked out next to the airflow one.
+""".format(
+    __file__
+)
+
+
+LOCAL_EXECUTOR_WARNING = """
+The test requires local executor. Please set AIRFLOW_CONFIG variable to '{}'
+and make sure you have a Postgres server running locally and
+airflow/airflow.db database created.
+
+You can create the database via these commands:
+'createuser root'
+'createdb airflow/airflow.db`
+
+"""
+
+
+class RetrieveVariables:
+    """
+    Retrieve environment variables from parent directory retriever - it should be
+    in the path ${AIRFLOW_SOURCES}/../../get_system_test_environment_variables.py
+    and it should print all the variables in form of key=value to the stdout
+    """
+
+    @staticmethod
+    @run_once
+    def retrieve_variables():
+        if os.path.isfile(ENV_FILE_RETRIEVER):
+            if os.environ.get("AIRFLOW__CORE__UNIT_TEST_MODE"):
+                raise Exception("Please unset the AIRFLOW__CORE__UNIT_TEST_MODE")
+            variables = subprocess.check_output([ENV_FILE_RETRIEVER]).decode("utf-8")
+            print("Applying variables retrieved")
+            for line in variables.split("\n"):
+                try:
+                    variable, key = line.split("=")
+                except ValueError:
+                    continue
+                print("{}={}".format(variable, key))
+                os.environ[variable] = key
+
+
+RetrieveVariables.retrieve_variables()
+
+
+@contextmanager
+def gcp_authentication(service_key: str, project_extra: Optional[str] = None):
+    """
+    Context manager for invoking GCP credentials. Can be used in test for
+    managing resources with specified key.
+
+    **Example**: ::
+        def setUp():
+            with gcp_authentication('power_service_key.json'):
+                create_bucket()
+
+
+        @gcp_authentication('power_service_key.json')
+        def setUp():
+            create_bucket()
+
+    :param service_key: name of the service key that will be used to provide credentials
+    :type service_key: str
+    :param project_extra: optional project extras
+    :type project_extra: str
+    """
+    auth = GcpAuthenticator(gcp_key=service_key, project_extra=project_extra)
```

Comment:
Do we need this class? I think this can be replaced with the code that is in GoogleCloudBaseHook: provide_gcp_credential_file_as_context method, but you need to extract logic to searate method. New method should take key_path and keyfile_dict as a parameter.

Reply:
So you suggest to setup a `GOOGLE_APPLICATION_CREDENTIALS` on enter and then remove it on exit of context manager?

Reply:
Yes. I think, we could also setup Airflow connection with gcp_key key - separate decorator.